test/testdrive/kafka-exactly-once-sink.td

```diff
@@ -228,24 +228,93 @@ $ kafka-verify format=avro sink=materialize.public.rt_binding_consistency_test_s
 
-! CREATE SINK compaction_test_sink FROM compaction_test_input
-  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'compaction-test-output-${testdrive.seed}'
-  CONSISTENCY TOPIC 'fail' CONSISTENCY FORMAT JSON
-  WITH (reuse_topic=true) FORMAT AVRO
-  USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}' ENVELOPE DEBEZIUM
-CONSISTENCY TOPIC and CONSISTENCY FORMAT not yet supported
+# Can't provide CONSISTENCY FORMAT and consistency_topic WITH option
+! CREATE SINK double_avro FROM unnamed_cols
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'double-avro'
+    CONSISTENCY TOPIC 'topicname'
+    WITH (consistency_topic='willfail')
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+Cannot specify consistency_topic and CONSISTENCY options simultaneously
+
+# Can't create sinks with JSON-encoded consistency topics
+! CREATE SINK double_avro FROM unnamed_cols
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'double-avro'
+    CONSISTENCY TOPIC 'consistency-double-avro' CONSISTENCY FORMAT JSON
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+CONSISTENCY FORMAT JSON not yet supported
+
+# Providing CONSISTENCY TOPIC without CONSISTENCY FORMAT will default to the sink's FORMAT
+# of the sink, if valid
+> CREATE SINK default_avro FROM unnamed_cols
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'default-avro'
+    CONSISTENCY TOPIC 'consistency-default-avro'
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+
+# Fail to default to JSON
+! CREATE SINK default_avro FROM unnamed_cols
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'default-avro'
+    CONSISTENCY TOPIC 'consistency-default-avro'
+  FORMAT JSON
+CONSISTENCY FORMAT JSON not yet supported
+
+> CREATE SINK double_avro FROM unnamed_cols
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'double-avro'
+    CONSISTENCY TOPIC 'consistency-double-avro' CONSISTENCY FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+
+$ kafka-verify format=avro sink=materialize.public.double_avro sort-messages=true
+{"before": null, "after": {"row": {"column1": 1, "b": 2, "column3": 3}}, "transaction": {"id": "0"}}
+
+> CREATE SINK json_avro FROM unnamed_cols
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'json-avro'
+    CONSISTENCY TOPIC 'consistency-json-avro' CONSISTENCY FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+  FORMAT JSON
+
+$ kafka-verify format=json sink=materialize.public.json_avro sort-messages=true key=false
+{"before": null, "after": {"row": {"column1": 1, "b": 2, "column3": 3}}, "transaction": {"id": "0"}}
+
+! CREATE SINK json_reuse_topic FROM unnamed_cols
+  INTO KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'json-reuse-topic'
+    WITH (reuse_topic=true)
+  FORMAT JSON
+CONSISTENCY FORMAT JSON not yet supported
```

Comment:
This one is a bit confusing, because the user didn't specify `CONSISTENCY FORMAT JSON` themselves. But I think the alternative would be to default to Avro consistency topic when using a JSON sink with `reuse_topic`. Which, I'm also not sure that's good.

Boy, this stuff is getting complicated... 😅 

The solution is probably future support for a JSON consistency topic. :trollface:

Reply:
Yeah, I agree!

The cleanest solution might be to not default to anything. While this is super explicit, it's not the most user-friendly experience. (This is what I originally did, but Philip pointed out that a default might be helpful).

It will get cleaner once we support a JSON consistency topic for sure!